src/app/tasks/task-details/task-details.component.ts

```diff
@@ -1,28 +1,116 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
-import { Task, emptyTask } from '../task.model';
-import { ActivatedRoute, Router } from '@angular/router';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Task, Option, emptyTask } from '../task.model';
+import { ActivatedRoute, Params, Router } from '@angular/router';
 import { TaskService } from '../task.service';
+import { FormControl, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { KeyValuePipe } from '@angular/common';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'app-task-details',
   standalone: true,
-  imports: [],
+  imports: [
+    ReactiveFormsModule,
+    KeyValuePipe
+  ],
   templateUrl: './task-details.component.html',
   styleUrl: './task-details.component.scss'
 })
-export class TaskDetailsComponent implements OnInit {
+export class TaskDetailsComponent implements OnInit, OnDestroy {
 
-  task: Task | null = null;
+  task: Task | null = emptyTask;
+  editMode: boolean = false;
+  taskTypes : Option[] = this.taskService.getTaskTypes();
+  taskStatuses : Option[] = this.taskService.getTaskStatuses();
+  routeParamsSubscription: Subscription = this.route.params.subscribe((params: Params) => {
+    this.task = this.taskService.getTask(+params['id']);
+  });
+
+  editTaskForm: FormGroup = new FormGroup({
+    title: new FormControl(),
+    description: new FormControl(),
+    type: new FormControl(),
+    status: new FormControl()
+  });
 
   onDeleted(taskId: number) {
     this.taskService.deleteTask(taskId);
     this.router.navigate(['tasks']);
   }
 
-  constructor(private router: Router, private route: ActivatedRoute, private taskService: TaskService) {}
+  onEdit() {
+    this.editMode = true;
```

Comment:
could be nice to maybe just give it the opposite value? (everything below is optional)
this.editMode = !this.editMode?
That way you can toggle between editMode, though if you decide to implement this then you have to reset the form to the initial values whenever you close the editMode